### PR TITLE
fix(translation): use unconditional xl() for dynamic strings, not xl_list_label

### PR DESCRIPTION
**Pull request by kojiromike**
## Summary

PRs #11429 (FHIR) and #11457 (reports) introduced calls to `xl_list_label()` on dynamic strings as a way to satisfy PHPStan's literal-string constraint. That was wrong: `xl_list_label()` is **not** a more permissive variant of `xl()` — it gates translation on the `translate_lists` global. Strings that aren't list_options labels (FHIR resource displays, message types, claim statuses, generic option labels) should not be conditionally translated based on a global meant for the list_options table.

This PR reverts all 14 misuses to unconditional `xl()` with scoped `@phpstan-ignore argument.type` comments (or, where the source turned out to be a known literal, drops the dynamic call entirely).

### Sites changed

**FHIR (9 sites, from #11429):**
- `src/Services/FHIR/FhirGoalService.php` (×2)
- `src/Services/FHIR/FhirLocationService.php` — `LocationService` hard-codes patient/user `name` to the literal `"Home Address"`, so this site translates the literal directly with no `@phpstan-ignore`.
- `src/Services/FHIR/FhirPatientService.php` (×2 — race/ethnicity, language)
- `src/Services/FHIR/FhirCareTeamService.php` (×2) — `role_title` is always populated by `CareTeamService::*` (`$row['role_title'] ?? $row['role']`), so the dead `?? xl($roleLabel)` fallback was dropped.
- `src/Services/FHIR/FhirMedicationRequestService.php`
- `src/Services/FHIR/DiagnosticReport/FhirDiagnosticReportLaboratoryService.php`

**Reports / billing (5 sites, from #11457):**
- `interface/billing/payment_master.inc.php` (the `$empty_name` call only — `$title_value` is a correct pre-existing list_options call and is left alone)
- `interface/reports/message_list.php` (×2 — msg_type, msg_status)
- `interface/reports/ippf_statistics.php`
- `library/ajax/billing_tracker_ajax.php` — `status` is now exposed as the raw enum (so JS comparisons against `'success'`/`'waiting'` keep working) and the translated form is exposed as a new `status_label` field.

**Billing tracker UI (follow-on for the `status_label` change):**
- `interface/billing/billing_tracker.php` — DataTables renderer updated to display `row.status_label` for the badge text while still comparing on `row.status` for the badge color.

### Why @phpstan-ignore is acceptable here

The dynamic strings come from the database (list_options titles, claim statuses, FHIR resource text, etc.). Translating them at runtime is the legacy "administrative translation" pattern. The follow-up issues track making this less hacky:

- #11496 — Tighten `xl_*` helpers to require literal-string
- #11497 — Introduce explicit `xl_admin*()` helper for administrative translation
- #11498 — Migrate dynamic callers to the new helper

Until those land, scoped `@phpstan-ignore argument.type` is the correct local fix.

## Test plan

- [x] `composer phpstan` reports no errors
- [x] `composer phpstan-baseline` produces no diff (no new baseline entries)
- [x] Pre-commit hooks pass

### CODE REVIEW

File: library/ajax/billing_tracker_ajax.php [hunk 1]
```diff
@@ -39,7 +39,9 @@
     $element->x12_partner_id = text($claim_file['x12_partner_id']);
     $element->x12_partner_name = text($claim_file['name']);
     $element->x12_filename = text($claim_file['x12_filename']);
-    $element->status = xl_list_label(is_string($claim_file['status'] ?? null) ? $claim_file['status'] : '');
+    $claimStatus = is_string($claim_file['status'] ?? null) ? $claim_file['status'] : '';
+    // @phpstan-ignore argument.type (dynamic claim status label from list_options)
+    $element->status = xl($claimStatus);
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply library/ajax/billing_tracker_ajax.php]: Good catch — `billing_tracker.php` does compare `data == 'success'`/`'waiting'` for the badge styling, so any translation here would break it (and the original `xl(...)` was already broken for non-English locales). Applied your suggestion: `status` is now the raw enum and the translated form is exposed as `status_label`. Fixed in 72deea1964.

File: src/Services/FHIR/FhirCareTeamService.php [hunk 2]
```diff
@@ -218,7 +218,8 @@ private function createRoleCodeableConcept($dataRecordProvider, $codeTypesServic
                 $codes = [
                     'code' => $roleMapping[$role],
                     'system' => self::CARE_TEAM_MEMBER_FUNCTION_SYSTEM,
-                    'description' => $dataRecordProvider['role_title'] ?? xl_list_label($roleLabel)
+                    // @phpstan-ignore argument.type (dynamic role label from list_options)
+                    'description' => $dataRecordProvider['role_title'] ?? xl($roleLabel)
                 ];
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply src/Services/FHIR/FhirCareTeamService.php]: Confirmed — `CareTeamService::getProviderRoles()` populates `role_title` as `$row['role_title'] ?? $row['role']`, so the `?? xl($roleLabel)` fallback is dead. Dropped the fallback (and the now-unused `$roleLabel` + `@phpstan-ignore`). Fixed in 72deea1964.

File: src/Services/FHIR/FhirCareTeamService.php [hunk 3]
```diff
@@ -418,7 +419,8 @@ public function populateRelatedPersonTeamMembers(FHIRCareTeam $careTeamResource,
                         $codes = [
                             'code' => $roleMapping[$role],
                             'system' => self::CARE_TEAM_MEMBER_FUNCTION_SYSTEM,
-                            'description' => $person['role_title'] ?? xl_list_label($roleLabel)
+                            // @phpstan-ignore argument.type (dynamic role label from list_options)
+                            'description' => $person['role_title'] ?? xl($roleLabel)
                         ];
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply src/Services/FHIR/FhirCareTeamService.php]: Same fix as the other call site — dropped the dead fallback. Fixed in 72deea1964.

File: src/Services/FHIR/FhirGoalService.php [hunk 4]
```diff
@@ -182,7 +182,8 @@ public function parseOpenEMRRecord($dataRecord = [], $encode = false)
                     $coding->setCode($cleanCode);
                     if (!empty($codeText)) {
                         // FIXED: Fix apostrophes in display text
-                        $cleanDisplay = str_replace('`', "'", xl_list_label($codeText));
+                        // @phpstan-ignore argument.type (dynamic code description from list_options)
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply src/Services/FHIR/FhirGoalService.php]: You're right — `$codeText` here comes from `CodeTypesService::lookup_code_description()` (code tables), not list_options. Updated the rationale comment. Fixed in 72deea1964.

File: src/Services/FHIR/FhirGoalService.php [hunk 5]
```diff
@@ -236,7 +237,8 @@ public function parseOpenEMRRecord($dataRecord = [], $encode = false)
                             $coding->setDisplay(UtilsService::createDataMissingExtension());
                         } else {
                             // FIXED: Fix apostrophes in target display text
-                            $cleanCodeText = str_replace('`', "'", xl_list_label($codeText));
+                            // @phpstan-ignore argument.type (dynamic code description from list_options)
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply src/Services/FHIR/FhirGoalService.php]: Same fix — updated the rationale to reference `lookup_code_description()`. Fixed in 72deea1964.

File: library/ajax/billing_tracker_ajax.php [hunk 6]
```diff
@@ -39,7 +39,11 @@
     $element->x12_partner_id = text($claim_file['x12_partner_id']);
     $element->x12_partner_name = text($claim_file['name']);
     $element->x12_filename = text($claim_file['x12_filename']);
-    $element->status = xl_list_label(is_string($claim_file['status'] ?? null) ? $claim_file['status'] : '');
+    $claimStatus = is_string($claim_file['status'] ?? null) ? $claim_file['status'] : '';
+    // Keep `status` as the raw enum so JS can compare against 'success'/'waiting'.
+    $element->status = $claimStatus;
+    // @phpstan-ignore argument.type (dynamic claim status label from list_options)
+    $element->status_label = xl($claimStatus);
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply library/ajax/billing_tracker_ajax.php]: Good catch — updated the DataTables renderer to display `row.status_label` for the badge text while still comparing on `row.status` for the badge color. Fixed in 0eaafca6d3.

File: src/Services/FHIR/FhirLocationService.php [hunk 7]
```diff
@@ -172,7 +172,8 @@ public function parseOpenEMRRecord($dataRecord = [], $encode = false)
         if (!empty($dataRecord['name'])) {
             $name = is_string($dataRecord['name']) ? $dataRecord['name'] : '';
             if ($dataRecord['type'] != 'facility') {
-                $name = xl_list_label($name);
+                // @phpstan-ignore argument.type (dynamic location name from list_options)
+                $name = xl($name);
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply src/Services/FHIR/FhirLocationService.php]: Confirmed in `LocationService::getAll()` — the SQL UNIONs hard-code `"Home Address" as name` for both the patient and user branches. Switched to `xl('Home Address')` directly and dropped the `@phpstan-ignore`. Fixed in 35ee622044.

File: interface/billing/billing_tracker.php [hunk 8]
```diff
@@ -73,14 +73,17 @@
...
-                                    data = '<span class="badge badge-success">' + jsText(data) + '</span>';
+                                    return '<span class="badge badge-success">' + jsText(label) + '</span>';
                                 } else if (data == 'waiting') {
-                                    data = '<span class="badge badge-info">' + jsText(data) + '</span>';
+                                    return '<span class="badge badge-info">' + jsText(label) + '</span>';
                                 } else {
-                                    data = '<span class="badge badge-warning">' + jsText(data) + '</span>';
+                                    return '<span class="badge badge-warning">' + jsText(label) + '</span>';
                                 }
                             }
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply interface/billing/billing_tracker.php:85]: Updated the PR description's "Sites changed" section to call out the `interface/billing/billing_tracker.php` UI change as a follow-on for the new `status_label` field.
